src/Catalyst.KBucket.Tests/SplitTest.cs

```diff
@@ -0,0 +1,135 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Catalyst.KBucket
+{
+    /// <summary>
+    ///   From https://github.com/tristanls/k-bucket/blob/master/test/split.js
+    /// </summary>
+    [TestClass]
+    public class SplitTest
+    {
+        [TestMethod]
+        public void OneContactDoesNotSplit()
+        {
+            var kBucket = new KBucket<Contact>
+            {
+                new Contact("a")
+            };
+            Assert.IsNull(kBucket.Root.Left);
+            Assert.IsNull(kBucket.Root.Right);
+            Assert.IsNotNull(kBucket.Root.Contacts);
+        }
+
+        [TestMethod]
+        public void MaxContactsPerNodeDoesNotSplit()
+        {
+            var kBucket = new KBucket<Contact>();
+            for (var i = 0; i < kBucket.ContactsPerBucket; ++i) kBucket.Add(new Contact(i));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'for' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)